src/app/zones/views/Zones.test.tsx

```diff
@@ -1,34 +1,67 @@
-import { mount } from "enzyme";
+import { screen, render } from "@testing-library/react";
 import { Provider } from "react-redux";
 import { MemoryRouter } from "react-router-dom";
 import { CompatRouter } from "react-router-dom-v5-compat";
 import configureStore from "redux-mock-store";
 
 import Zones from "./Zones";
 
+import type { RootState } from "app/store/root/types";
 import zonesURLs from "app/zones/urls";
-import { rootState as rootStateFactory } from "testing/factories";
+import {
+  authState as authStateFactory,
+  zone as zoneFactory,
+  zoneState as zoneStateFactory,
+  rootState as rootStateFactory,
+  user as userFactory,
+  userState as userStateFactory,
+} from "testing/factories";
 
 const mockStore = configureStore();
 
 describe("Zones", () => {
+  let initialState: RootState;
+
+  beforeEach(() => {
+    initialState = rootStateFactory({
+      user: userStateFactory({
+        auth: authStateFactory({
+          user: userFactory({ is_superuser: true }),
+        }),
+      }),
+      zone: testZones,
+    });
+  });
+
+  const testZones = zoneStateFactory({
+    errors: {},
+    loading: false,
+    loaded: true,
+    items: [
+      zoneFactory({
+        id: 1,
+        name: "zone-name",
+      }),
+    ],
+  });
+
   [
     {
-      component: "ZonesList",
+      component: "Availability zones",
       path: zonesURLs.index,
     },
     {
-      component: "ZoneDetails",
+      component: /Availability Zone/i,
       path: zonesURLs.details({ id: 1 }),
     },
     {
-      component: "NotFound",
-      path: "/not/a/path",
+      component: "Zone not found",
+      path: zonesURLs.details({ id: 2 }),
     },
   ].forEach(({ component, path }) => {
     it(`Displays: ${component} at: ${path}`, () => {
-      const store = mockStore(rootStateFactory());
-      const wrapper = mount(
+      const store = mockStore(initialState);
+      render(
```

Comment:
The main reason this isn't working is that for these tests to work with React Testing Library the components need to be contained inside the matching route pattern e.g. for a zone it would be `/zone/:id` this is so the path params, in this case `:id` can be captured.

Inside this test you could add the `<Routes>` and `<Route>` components, but a better way would be to use the `renderWithBrowserRouter` function and pass the correct `routePattern` for each path. That could look something like this:

```TypeScript
[
  {
    label: ZoneListLabel.List,
    path: zonesURLs.index,
    pattern: zonesURLs.index,
  },
  {
    label: ZoneDetailsLabel.Details,
    path: zonesURLs.details({ id: 1 }),
    pattern: zonesURLs.details(null, true),
  },
  {
    label: NotFoundLabel.Title,
    path: `${zonesURLs.index}/not/a/path`,
    pattern: `${zonesURLs.index}/*`,
  },
].forEach(({ label, path, pattern }) => {
  it(`Displays: ${label} at: ${path}`, () => {
    renderWithBrowserRouter(<Zones />, {
      wrapperProps: { routePattern: pattern, state },
      route: path,
    });
    expect(screen.getByLabelText(label)).toBeInTheDocument();
  });
});
```

I've pushed a branch as a full example of how I got this test to pass:
https://github.com/canonical-web-and-design/maas-ui/commit/41066b2c4c501856c2ef3905037f0ce5f3d97325

You could also look at the Tags component as a reference: https://github.com/canonical-web-and-design/maas-ui/blob/main/src/app/tags/views/Tags.test.tsx#L41.